### PR TITLE
feat(platform): Add javascript-nuxt platform in FE

### PR DESCRIPTION
**Pull request by s1gr1d**
Frontend part for the Nuxt onboarding.

Backend part: https://github.com/getsentry/sentry/pull/76851

### CODE REVIEW

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+      link: 'https://docs.sentry.io/platforms/javascript/guides/nuxt/tracing/',
+    },
+    {
+      id: 'session-replay',
+      name: t('Session Replay'),
+      description: t(
+        'Get to the root cause of an error or latency issue faster by seeing all the technical details related to that issue in one visual replay on your web application.'
+      ),
+      link: 'https://docs.sentry.io/platforms/javascript/guides/nuxt/session-replay/',
+    },
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:275]: We have a open [PR](https://github.com/getsentry/sentry/pull/78461/files) that removes the product specific next steps.
```suggestion
```

**Review by ArthurKnaus — COMMENTED** [review]
You will also want to add it to https://github.com/getsentry/sentry/blob/master/static/app/data/platformCategories.tsx

File: static/app/components/platformPicker.tsx [hunk 2]
```diff
@@ -107,13 +107,17 @@ class PlatformPicker extends Component<PlatformPickerProps, State> {
     const filtered = tempSelectablePlatforms
       .filter(this.state.filter ? subsetMatch : categoryMatch)
       .sort((a, b) => {
-        if (startsWithPunctuation(a.name) && !startsWithPunctuation(b.name)) {
+        // Treat Nuxt as 'Vue' to sort it alongside Nuxt
+        const nameA = a.id === 'javascript-nuxt' ? 'Vue' : a.name;
+        const nameB = b.id === 'javascript-nuxt' ? 'Vue' : b.name;
+
+        if (startsWithPunctuation(nameA) && !startsWithPunctuation(nameB)) {
```
[the remark this answers is not in the document]
Comment (s1gr1d) [reply static/app/components/platformPicker.tsx:114]: I agree that it seems weird in the code but when looking at the product selection UI it's actually more weird if everything is sorted alphabetically :D

The meta-frameworks are always alongside their client-side framework/library (svelte - sveltekit, solid - solidstart, nuxt - react, svelte - sveltekit, ...) and the first thing a human would do is pattern-match and I think the pattern-matching here is more on the framework level than on the alphabeth level. What do you think?

[the remark this answers is not in the document]
Comment (ArthurKnaus) [reply static/app/components/platformPicker.tsx:114]: K, I understand the motivation. Still it feels a bit weird to me to break out of the sorting especially as we do not group other platforms (the ones you mentioned are grouped coincidentally only on the browser tab).

To be clear: I'm fine with merging it. Just made me question the whole (current) logic. 😅

**Comment by ArthurKnaus** [comment]
Code looks good 👍 Will leave content approval to @lforst and @chargome

**Review by chargome — APPROVED** [review]

**Review by scttcper — APPROVED** [review]

File: static/app/components/platformPicker.tsx [hunk 3]
```diff
@@ -107,13 +107,18 @@ class PlatformPicker extends Component<PlatformPickerProps, State> {
     const filtered = tempSelectablePlatforms
       .filter(this.state.filter ? subsetMatch : categoryMatch)
       .sort((a, b) => {
-        if (startsWithPunctuation(a.name) && !startsWithPunctuation(b.name)) {
+        // Treat Nuxt as 'Vue' to sort it alongside Nuxt
+        // Other related frameworks are sorted next to each other automatically (solid - solidstart, svelte - sveltekit, ...)
```
Comment (priscilawebdev) [inline static/app/components/platformPicker.tsx:111]: I recently updated the platform list alphabetically. You can check the changes in [this PR](https://github.com/getsentry/sentry/pull/74832). With these updates, Nuxt won't be next to Next. Why did we decide to do that?

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+            },
+          ],
+        },
+        {
+          description: (
+            <Fragment>
+              {tct(
+                'For the server, create a [codeFile:sentry.server.config.ts] file in your project root and initialize the Sentry SDK:',
+                {codeFile: <code />}
+              )}
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:192]:
```suggestion
 <p>
  {tct(
                'For the server, create a [codeFile:sentry.server.config.ts] file in your project root and initialize the Sentry SDK:',
                {codeFile: <code />}
              )}
 </p>
```

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+const docs: Docs = {
+  onboarding,
+  feedbackOnboardingNpm: feedbackOnboarding,
+  replayOnboarding,
+  customMetricsOnboarding: getJSMetricsOnboarding({getInstallConfig}),
+  crashReportOnboarding,
+};
+
+const StyledAlert = styled(Alert)`
+  margin-top: ${space(2)};
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:367]:
```suggestion
```

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+      description: t(
+        "This snippet contains an intentional error and can be used as a test to make sure that everything's working as expected."
+      ),
+      configurations: [
+        {
+          code: [
+            {
+              label: 'JavaScript',
+              value: 'javascript',
+              language: 'html',
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:241]:
```suggestion
              language: 'javascript',
```

File: static/app/gettingStartedDocs/javascript/nuxt.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
+import {renderWithOnboardingLayout} from 'sentry-test/onboarding/renderWithOnboardingLayout';
+import {screen} from 'sentry-test/reactTestingLibrary';
+import {textWithMarkupMatcher} from 'sentry-test/utils';
+
+import {ProductSolution} from 'sentry/components/onboarding/productSelection';
+
+import docs from './nuxt';
+
+describe('javascript-nuxt onboarding docs', function () {
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.spec.tsx:9]: great test 👏

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+          description: tct(
+            'Add the Sentry Nuxt module in your [code:nuxt.config.ts] file:',
+            {code: <code />}
+          ),
+          code: [
+            {
+              label: 'TypeScript',
+              value: 'typescript',
+              language: 'typescript',
+              code: getNuxtModuleSnippet(),
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:168]:
```suggestion
              language: 'typescript',
              filename: 'nuxt.config.ts',
              code: getNuxtModuleSnippet(),
```

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 9]
```diff
@@ -0,0 +1,371 @@
...
+                  }
+                )}
+              </StyledAlert>
+            </Fragment>
+          ),
+          code: [
+            {
+              label: 'TypeScript',
+              value: 'typescript',
+              language: 'typescript',
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:210]:
```suggestion
              language: 'typescript',
              filename: 'sentry.server.config.ts',
```

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+          description: tct(
+            'For the client, create a [codeFile:sentry.client.config.ts] file in your project root and initialize the Sentry SDK:',
+            {codeFile: <code />}
+          ),
+          code: [
+            {
+              label: 'TypeScript',
+              value: 'typescript',
+              language: 'typescript',
+              code: getSdkClientSetupSnippet(params),
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:182]:
```suggestion
              language: 'typescript',
              filename: 'sentry.client.config.ts',
              code: getSdkClientSetupSnippet(params),
```

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 11]
```diff
@@ -0,0 +1,371 @@
...
+  // dsn: useRuntimeConfig().public.sentry.dsn,
+  dsn: "${params.dsn.public}",${
+    params.isReplaySelected
+      ? `
+  integrations: [Sentry.replayIntegration(${getReplayConfigOptions(params.replayOptions)})],`
+      : ''
+  }${
+    params.isPerformanceSelected
+      ? `
+  // Performance Monitoring
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:52]:
```suggestion
  // Tracing
```

File: static/app/components/platformPicker.tsx (hunk 3 above)
Comment (s1gr1d) [reply static/app/components/platformPicker.tsx:111]: See the discussion above :)
https://github.com/getsentry/sentry/pull/78467#discussion_r1784441878

File: static/app/gettingStartedDocs/javascript/nuxt.tsx (hunk 6 above)
Comment (s1gr1d) [reply static/app/gettingStartedDocs/javascript/nuxt.tsx:241]: The code is Vue code so I have to use html here to get correct syntax highlighting. But I changed the label and value to vue.

File: static/app/components/platformPicker.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/components/platformPicker.tsx:111]: Yes I saw that afterward, but I am still not convinced 😬 let discuss that internally

File: static/app/gettingStartedDocs/javascript/nuxt.tsx [hunk 12]
```diff
@@ -0,0 +1,374 @@
...
+        value: 'pnpm',
+        language: 'bash',
+        code: `pnpm add @sentry/nuxt`,
+      },
+    ],
+  },
+];
+
+const onboarding: OnboardingConfig = {
+  introduction: MaybeBrowserProfilingBetaWarning,
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.tsx:143]: one last thing. don't we want to add this here? 
```suggestion
 introduction: params => (
    <Fragment>
      <MaybeBrowserProfilingBetaWarning {...params} />
      <p>
        {tct('In this quick guide you’ll use [strong:npm], [strong:yarn] or [strong:pnpm] to set up:', {
          strong: <strong />,
        })}
      </p>
    </Fragment>
  ),
```

File: static/app/gettingStartedDocs/javascript/nuxt.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+        ProductSolution.PERFORMANCE_MONITORING,
+        ProductSolution.SESSION_REPLAY,
+      ],
+    });
+
+    expect(screen.queryAllByText(textWithMarkupMatcher(/tracesSampleRate/))).toHaveLength(
+      2
+    ); // client and server
+    expect(
+      screen.queryByText(textWithMarkupMatcher(/replaysSessionSampleRate/))
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.spec.tsx:40]:
```suggestion
      screen.getByText(textWithMarkupMatcher(/replaysSessionSampleRate/))
```

`queryBy` is usually used when we do not expect the dom to have the element 🙂

File: static/app/gettingStartedDocs/javascript/nuxt.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    ).toBeInTheDocument(); // only client
+    expect(
+      screen.queryByText(textWithMarkupMatcher(/replaysOnErrorSampleRate/))
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.spec.tsx:43]:
```suggestion
      screen.getByText(textWithMarkupMatcher(/replaysOnErrorSampleRate/))
```

File: static/app/gettingStartedDocs/javascript/nuxt.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+  it('displays sample rates by default', () => {
+    renderWithOnboardingLayout(docs, {
+      selectedProducts: [
+        ProductSolution.ERROR_MONITORING,
+        ProductSolution.PERFORMANCE_MONITORING,
+        ProductSolution.SESSION_REPLAY,
+      ],
+    });
+
+    expect(screen.queryAllByText(textWithMarkupMatcher(/tracesSampleRate/))).toHaveLength(
```
Comment (priscilawebdev) [inline static/app/gettingStartedDocs/javascript/nuxt.spec.tsx:36]:
```suggestion
    expect(screen.getAllByText(textWithMarkupMatcher(/tracesSampleRate/))).toHaveLength(
```

**Review by priscilawebdev — APPROVED** [review]
🚀

**Pull request merged**